Fix session pickling

de-vri-es opened the pull request:
This PR fixes session pickling. The file to dump the pickled session to must be opened in binary mode, or it fails with an exception: `TypeError: must be str, not bytes`. This probably broke with python 3. 

Additionally, this PR improves error handling a bit: when unexpected errors occur during authentication, the full exception traceback is added to the event log. Similarly, when unpickling fails other than with a FileNotFoundError, a traceback is logged as warning and an empty session is used.

Finally, it replaces the use of the deprecated `xbmc.translatePath()` with `xbmcvfs.translatePath()`.

de-vri-es committed:
[plugin.video.curisositystream] Gracefully handle all unpickling errors.

de-vri-es committed:
[plugin.video.curiositystream] Replace deprecated xbmc.translatePath.

de-vri-es committed:
[plugin.video.curiositystream] Open session file in binary mode.

de-vri-es committed:
[plugin.video.curiositystream] Log full exception on internal errors.

conor-f commented:
Can confirm this fix works and is necessary for the plugin to run.

Installed this earlier and it was just throwing errors, applied this patch and it worked. Please merge as it is in the official repos.

ABuijert commented:
Can also confirm. For almost a year my password/user failed. With this commit it is working again.

m0n5t3r commented:
heh, just fixed this on my own before finding this; the lack of response makes me think this repo is abandoned, you'd be better off submitting the PR to upstream (I was about to send my own PR, but yours is more polished and I have no idea what to do with versions anyway, as I started looking into how kodi add-ons work 1h ago, and spent about half of the time locating places to send the PR with the fix :joy:)

de-vri-es commented:
Yeah, fair enough. I wanted to try this first anyway, because it felt like the right thing to do. But I will submit this upstream soon :)

m0n5t3r commented:
btw, there's also an `EOFError` you can get if the session file is empty - I was getting it because the initial session save failed due to the bytes error, leaving an empty file

edit: nevermind, you catch them all, it's probably better in this case :)
edit again: ah, that's just in the saving code; I guess if the loading code crashes for some reason it is okay to let it show an error and have the user investigate (I'm still sleeping apparently)

de-vri-es commented:
> edit: nevermind, you catch them all, it's probably better in this case :)
> edit again: ah, that's just in the saving code; I guess if the loading code crashes for some reason it is okay to let it show an error and have the user investigate (I'm still sleeping apparently)

It actually is in the loading code ;P

What I changed is that only FileNotFoundErrors are silently ignored where before it was all IOErrors. Other errors are now just added to the event log for debugging purposes, but not shown to the user.

de-vri-es commented:
New PR opened here: https://github.com/xbmc/repo-plugins/pull/3955

de-vri-es commented:
It is now merged into the official repositories. Closing this PR.
